src/plugins/explore/public/components/visualizations/bar/bucket_options.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { i18n } from '@osd/i18n';
+import { EuiFormRow, EuiSelect, EuiFieldNumber } from '@elastic/eui';
+import React from 'react';
+import { BarChartStyleControls } from './bar_vis_config';
+import { AggregationType, TimeUnit, BucketOptions } from '../types';
+import { getAggregationnType, getTimeUnits } from '../utils/collections';
+import { useDebouncedValue } from '../utils/use_debounced_value';
+import { StyleAccordion } from '../style_panel/style_accordion';
+
+interface BucketVisOptionsProps {
+  styles: BarChartStyleControls['bucket'];
+  bucketType: 'time' | 'num' | 'single';
+  onChange: (styles: BarChartStyleControls['bucket']) => void;
+}
+
+export const BucketVisOptions = ({ styles, onChange, bucketType }: BucketVisOptionsProps) => {
+  const updateBucketOption = <K extends keyof BucketOptions>(key: K, value: BucketOptions[K]) => {
+    onChange({
+      ...styles,
+      [key]: value,
+    });
+  };
+
+  const [localBucketSize, setLocalBucketSize] = useDebouncedValue<number | undefined>(
+    styles?.bucketSize,
+    (val) => updateBucketOption('bucketSize', val)
+  );
+
+  const [localBucketCount, setLocalBucketCount] = useDebouncedValue<number | undefined>(
+    styles?.bucketCount,
+    (val) => updateBucketOption('bucketCount', val)
+  );
+
+  const labelType = getAggregationnType();
+  const timeUnits = getTimeUnits();
+
+  return (
+    <StyleAccordion
+      id="bucketSection"
+      accordionLabel={i18n.translate('explore.stylePanel.bar.tabs.bucket', {
+        defaultMessage: 'Bucket',
+      })}
+      initialIsOpen={true}
+      data-test-subj="bucketPanel"
+    >
+      {bucketType !== 'single' && (
+        <EuiFormRow
+          label={i18n.translate('explore.stylePanel.bar.bucket.type', {
+            defaultMessage: 'Type',
+          })}
+        >
+          <EuiSelect
+            compressed
+            value={styles?.aggregationType}
+            onChange={(e) =>
+              updateBucketOption('aggregationType', e.target.value as AggregationType)
+            }
+            options={labelType}
+          />
+        </EuiFormRow>
+      )}
+
+      {bucketType === 'time' && (
+        <EuiFormRow
+          label={i18n.translate('explore.stylePanel.bar.bucket.timeUnit', {
+            defaultMessage: 'Time interval selection',
```

Comment:
What about simply call it:
```suggestion
            defaultMessage: 'Interval',
```